lua/lspconfig/server_configurations/kotlin_language_server.lua

```diff
@@ -24,6 +24,9 @@ return {
     filetypes = { 'kotlin' },
     root_dir = util.root_pattern(unpack(root_files)),
     cmd = { bin_name },
+    init_options = {
+      storagePath = vim.loop.os_homedir(),
```

Comment:
why server will use home dir storage data ?

Reply:
I think using the project workspace is the right place for it but I could figure out a way to get the project root path here. The `root_pattern` returns a function. Do you know if I can set the project root here?

Reply:
For the context the cache file is a single sqlite DB file.

Reply:
root_pattern return a function then pass current buffer name to it and invoke it again.

Reply:
I did it using vim.fn.expand('%:p:h').
First I tried to use `vim.api.nvim_buf_get_name(0)`. But root_pattern expects a directory. So I used the fn.expand because it's the short & clear to read. I could not find a one liner in lua APIs to get the current directory.